### DISCUSSION TITLE
Why is `server.url` not intended for use in production?

### DISCUSSION THREAD

**Discussion by yukukotani** [ionic-team/capacitor#5075] [Q&A]
[The docs](https://capacitorjs.com/docs/config) say that `server.url` is for live-reload server.
```ts
/**
* Load an external URL in the Web View.
*
* This is intended for use with live-reload servers.
*
* **This is not intended for use in production.**
*
* @since 1.0.0
*/
url?: string;
```

Why this shouldn't be used in production? Any risk (like security issue)?

**Comment by webuniverseio** [comment]
@Monchi from my experience it fixes more problems that it introduce. I can see how others are thinking in the same direction https://github.com/ionic-team/capacitor/issues/4122 . We definitely need some clarify on that

**Reply by webuniverseio** [reply]
This is where it was introduced I asked for clarification https://github.com/ionic-team/capacitor/pull/3756#pullrequestreview-815430463

Original addition of a confusing comment https://github.com/ionic-team/capacitor/pull/3756/commits/7861b808e034f32068c4df03400f334786e8531f#r756466909

**Reply by yukukotani** [reply]
@webuniverseio Thanks!

I've published a capacitor app with `server.url` half-year ago, and it's going well on both App Store and Google Play.

The only problem I faced is updating capacitor plugins. When I update plugins in the web bundle, users still have an old native app with old plugins, therefore the version difference sometimes causes an error.

**Reply by buschtoens** [reply]
We also have a production app using that feature. It's going great.

I also subscribe to the plugin issue. We're dealing with that by adding a `version` property to the plugins, which we use as a type discriminator to select the right TS interface. This allows us to use the type system for versioning and eliminate many issues.

Eventually though, we force users to upgrade their app, if they are unreasonably far out of date and we've removed support for certain plugin versions.

**Reply by dylanvdmerwe** [reply]
How does offline work with this? Surely the app won't load if there is no internet connection? Or do you let the SW cache everything?

Also how is Apple with approving an website "wrapper" like this?

**Reply by yukukotani** [reply]
@buschtoens 
We're doing force upgrading too. We do this in the native layer instead of the web layer because sometimes WebView crashes before showing upgrade instruction.

@dylanvdmerwe 
Our app needs an internet connection, so we've not dealt with that problem. As you say, the app won't load if there is no internet connection, so we're showing an error page on the native layer.

We've not done anything special to get Apple's approval with WebView wrapper. We've just added native-specific features such as push notifications.

**Reply by dylanvdmerwe** [reply]
Problem is capacitor is not being injected (on Android) when the site is served by a service worker. [This ](https://github.com/ionic-team/capacitor/issues/1655#issuecomment-579229390)no longer works. Effectively preventing me me from using [native plugins](https://github.com/ionic-team/capacitor/issues/5278). 

How are you getting your app to work? Do you have a service worker? Raised [here](https://github.com/ionic-team/capacitor/issues/5278).
